aggregator/handler/config.go

```diff
@@ -69,34 +69,36 @@ func (c FlushHandlerConfiguration) NewHandler(
 		sharderRouters = make([]SharderRouter, 0, len(c.Handlers))
 	)
 	for _, hc := range c.Handlers {
-		switch hc.Type {
+		if err := hc.Validate(); err != nil {
+			return nil, err
+		}
+		if hc.DynamicBackend != nil {
+			sharderRouter, err := hc.DynamicBackend.NewSharderRouter(
+				cs,
+				instrumentOpts,
+			)
+			if err != nil {
+				return nil, err
+			}
+			sharderRouters = append(sharderRouters, sharderRouter)
+			continue
+		}
+		switch hc.StaticBackend.Type {
```

Comment:
Can we move this `switch` statement to the `NewSharderRouter` method of `StaticBackend` so it's a bit cleaner?

Reply:
I could, but it may create a handler or a SharderRouter depends on the type, so I'll need a `NewHandlerOrSharderRouter` function and return 2 arguments, then do the append here, which is also messy, up to you.

Reply:
Hm we could pass in the handlers and routers, do the append inside the method, and return them, but yeah don't feel strongly about it, let's leave it as is for now.